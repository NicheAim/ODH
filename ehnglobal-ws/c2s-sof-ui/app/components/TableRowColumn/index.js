/**
 *
 * TableRowColumn
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const TableRowColumn = styled.div`
  color: #000;
  font-family: "Arial", sans-serif;
  font-style: normal;
  font-size: 12px;
  font-weight: 400;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-all;
  padding: 5px;
`;

TableRowColumn.propTypes = {
  children: PropTypes.node,
};

export default TableRowColumn;
