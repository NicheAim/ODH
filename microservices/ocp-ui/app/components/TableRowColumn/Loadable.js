/**
 *
 * Asynchronously loads the component for TableRowColumn
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
