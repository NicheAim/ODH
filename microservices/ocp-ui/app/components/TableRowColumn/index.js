/**
 *
 * TableRowColumn
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const textDecorationLineValues = ['blink', 'inherit', 'initial', 'line-through', 'none', 'overline', 'underline', 'unset'];

const TableRowColumn = styled.div`
  color: #000;
  font-family: "Arial", sans-serif;
  font-style: normal;
  font-size: 12px;
  font-weight: 400;
  line-height: 200%;
  text-align: left;
  margin: 0;
  word-break: break-word;
  padding: 5px;
  text-decoration-line: ${({ textDecorationLine }) => textDecorationLine};
`;

TableRowColumn.propTypes = {
  children: PropTypes.node,
  textDecorationLine: PropTypes.oneOf(textDecorationLineValues),
};
TableRowColumn.defaultProps = {
  textDecorationLine: 'none',
};
export default TableRowColumn;
