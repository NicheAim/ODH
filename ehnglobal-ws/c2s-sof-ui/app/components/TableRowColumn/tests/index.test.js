import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import TableRowColumn from '../index';

configure({ adapter: new Adapter() });

describe('<TableRowColumn />', () => {
  it('should match snapshot', () => {
    // Act
    const renderedComponent = shallow(
      (<TableRowColumn>Row</TableRowColumn>),
    );

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
});
