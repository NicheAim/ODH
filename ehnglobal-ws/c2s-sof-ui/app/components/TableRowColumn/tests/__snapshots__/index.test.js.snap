// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRowColumn /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRowColumn__TableRowColumn>
    Row
  </TableRowColumn__TableRowColumn>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Row",
      "className": "TableRowColumn__TableRowColumn-VzNyU hoCwJx",
    },
    "ref": null,
    "rendered": "Row",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Row",
        "className": "TableRowColumn__TableRowColumn-VzNyU hoCwJx",
      },
      "ref": null,
      "rendered": "Row",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
